### PR TITLE
[Forwardport] Use batches and direct queries to fix sales address upgrade

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description (*)
This is forwardport of https://github.com/magento/magento2/pull/19098 to 2.3-develop

Magento Commerce compatible fix for sales address upgrade script.

### Fixed Issues (if relevant)
1. magento/magento2#16570: enhance performance on large catalog [(comment)](https://github.com/magento/magento2/pull/16570#issuecomment-429830198)
2. https://github.com/magento/magento2/pull/18945

### Manual testing scenarios (*)
Magento Commerce compatible fix for sales address upgrade script.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @ihor-sviziev. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by ihor-sviziev** [comment]
Hi,
I did the same changes as in https://github.com/magento/magento2/pull/19098, that I really tested, wanted to approve, but fix for 2.3-develop should be merged first.

**Review by slavvka — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @slavvka, thank you for the review.
[ENGCOM-4071](https://jira.corp.magento.com/browse/ENGCOM-4071) has been created to process this Pull Request

**Comment by p-bystritsky** [comment]
@ihor-sviziev there is not such column as "orderAddressId" in "sales_order_address" table, used in 'where' clause in https://github.com/magento/magento2/pull/19098/files#diff-0815805b2d40fcf0d22ebd639e0d73c4R247. Which will lead to an error during update.

File: app/code/Magento/Sales/Setup/Patch/Data/FillQuoteAddressIdInSalesOrderAddress.php [hunk 1]
```diff
@@ -131,18 +84,107 @@ public static function getDependencies()
...
+            ->where('address_type = ?', $addressType);
+
+        $quoteAddresses = $quoteConnection->fetchAssoc($query);
+
+        foreach ($orderAddresses as $orderAddress) {
+            $bind = [
+                'quote_address_id' => $quoteAddresses[$orderAddress['quote_id']]['address_id'] ?? null,
+            ];
+            $where = [
+                'orderAddressId' => $orderAddress['entity_id']
```
Comment (ihor-sviziev) [inline app/code/Magento/Sales/Setup/Patch/Data/FillQuoteAddressIdInSalesOrderAddress.php]:
```suggestion
                'entity_id' => $orderAddress['entity_id']
```

**Comment by ihor-sviziev** [comment]
As we discussed with @p-bystritsky  in slack - change with entity_id fixed SQL error, but in general - data is incorrect. Need to re-check.

**Comment by ihor-sviziev** [comment]
Hi @p-bystritsky,
I found & fixed root of the issue. Please re-check it. Also added the same change to backport in 2.2-develop branch.
Thank you!

**Pull request merged**